jmh/src/main/java/com/uber/nullaway/jmh/NullawayJavac.java

```diff
@@ -0,0 +1,158 @@
+package com.uber.nullaway.jmh;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import javax.tools.DiagnosticListener;
+import javax.tools.JavaCompiler;
+import javax.tools.JavaFileObject;
+import javax.tools.SimpleJavaFileObject;
+import javax.tools.StandardJavaFileManager;
+import javax.tools.ToolProvider;
+
+/** Code to run Javac with NullAway enabled. */
+public class NullawayJavac {
+
+  private static int getJavaRuntimeVersion() {
+    String version = System.getProperty("java.version");
+    if (version.startsWith("1.")) {
+      version = version.substring(2, 3);
+    } else {
+      int dot = version.indexOf(".");
+      if (dot != -1) {
+        version = version.substring(0, dot);
+      }
+    }
+    return Integer.parseInt(version);
+  }
+
+  public static void main(String[] args) throws Exception {
```

Comment:
Maybe a javadoc here explaining that (assuming I understand this correctly) this is not the actual entry point to the benchmark, but that it's run by the jmh gradle plugin in such and such way...?